ibis/expr/tests/snapshots/test_format/test_memoize_filtered_tables_in_join/repr.txt

```diff
@@ -17,13 +17,15 @@ r2 := Filter[r1]
 r3 := Filter[r1]
   r1.kind == 'bar'
 
-r4 := SelfReference[r3]
+r4 := SelfReference[r2]
```

Comment:
These are rather ugly. We could just omit rendering self references and place a `!r2` or something prefixed like that in the `JoinChain` repr.

Reply:
How about `&` if we're introducing new "syntax" in the repr?